--- a/externals/oaknut/include/oaknut/code_block.hpp
+++ b/externals/oaknut/include/oaknut/code_block.hpp
@@ -25,7 +25,7 @@
 
 namespace oaknut {
 
-#if defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR
+#if defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR && !defined(PVZ2_DYNARMIC_FORCE_NONTXM_JIT)
 namespace detail {
 
 struct ReusableJitRegion {
@@ -68,7 +68,7 @@
 #if defined(_WIN32)
         m_memory = (std::uint32_t*)VirtualAlloc(nullptr, size, MEM_COMMIT, PAGE_EXECUTE_READWRITE);
 #elif defined(__APPLE__)
-#    if TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR
+#    if TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR && !defined(PVZ2_DYNARMIC_FORCE_NONTXM_JIT)
         auto& reusable_region = detail::reusable_jit_region();
         if (reusable_region.memory != nullptr) {
             if (reusable_region.in_use || reusable_region.size != size)
@@ -123,14 +123,14 @@
 
 #if defined(_WIN32)
         if (m_memory == nullptr)
-#elif defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR
+#elif defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR && !defined(PVZ2_DYNARMIC_FORCE_NONTXM_JIT)
         if (m_memory == nullptr || m_wmemory == nullptr)
 #else
         if (m_memory == MAP_FAILED)
 #endif
             throw std::bad_alloc{};
 
-#if !(defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR)
+#if !(defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR && !defined(PVZ2_DYNARMIC_FORCE_NONTXM_JIT))
         m_wmemory = m_memory;
 #endif
     }
@@ -143,7 +143,7 @@
 #if defined(_WIN32)
         VirtualFree((void*)m_memory, 0, MEM_RELEASE);
 #else
-#    if defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR
+#    if defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR && !defined(PVZ2_DYNARMIC_FORCE_NONTXM_JIT)
         auto& reusable_region = detail::reusable_jit_region();
         if (reusable_region.memory == m_memory && reusable_region.writable_memory == m_wmemory) {
             reusable_region.in_use = false;
@@ -179,7 +179,7 @@
     {
 #if defined(__APPLE__) && !TARGET_OS_IPHONE
         pthread_jit_write_protect_np(1);
-#elif defined(__APPLE__) && TARGET_OS_IPHONE && TARGET_OS_SIMULATOR
+#elif defined(__APPLE__) && TARGET_OS_IPHONE && (TARGET_OS_SIMULATOR || defined(PVZ2_DYNARMIC_FORCE_NONTXM_JIT))
         mprotect(m_memory, m_size, PROT_READ | PROT_EXEC);
 #elif defined(__NetBSD__) || defined(__OpenBSD__)
         mprotect(m_memory, m_size, PROT_READ | PROT_EXEC);
@@ -190,7 +190,7 @@
     {
 #if defined(__APPLE__) && !TARGET_OS_IPHONE
         pthread_jit_write_protect_np(0);
-#elif defined(__APPLE__) && TARGET_OS_IPHONE && TARGET_OS_SIMULATOR
+#elif defined(__APPLE__) && TARGET_OS_IPHONE && (TARGET_OS_SIMULATOR || defined(PVZ2_DYNARMIC_FORCE_NONTXM_JIT))
         mprotect(m_memory, m_size, PROT_READ | PROT_WRITE);
 #elif defined(__NetBSD__) || defined(__OpenBSD__)
         mprotect(m_memory, m_size, PROT_READ | PROT_WRITE);
@@ -199,7 +199,7 @@
 
     void detach_jit_server()
     {
-#if defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR
+#if defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR && !defined(PVZ2_DYNARMIC_FORCE_NONTXM_JIT)
         if (m_should_detach_jit_server) {
             m_should_detach_jit_server = false;
             detail::detach_jit_server();
@@ -258,7 +258,7 @@
     std::uint32_t* m_memory = nullptr;
     std::uint32_t* m_wmemory = nullptr;
     std::size_t m_size = 0;
-#if defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR
+#if defined(__APPLE__) && TARGET_OS_IPHONE && !TARGET_OS_SIMULATOR && !defined(PVZ2_DYNARMIC_FORCE_NONTXM_JIT)
     bool m_should_detach_jit_server = false;
 #endif
 };
